cpu/apic: multiple trivial cleanups

00xc opened the pull request:
* Add documentation for several functions.
* Remove the `pub` qualifier from functions that are not used outside the APIC module.
* Clean up formatting in some comments
* Make a few variables immutable.

msft-jlange commented:
I'm not sure we want to spend much effort converting the number of spaces after periods in comments.  The VIM editor, for example (which I use) defaults to two spaces after a period when joining two comment lines together.  I don't object to changes to these comments but we could easily wind up wasting a lot of time with comment editing if we try to get strict about requiring specific rules.

00xc commented:
> I'm not sure we want to spend much effort converting the number of spaces after periods in comments. The VIM editor, for example (which I use) defaults to two spaces after a period when joining two comment lines together. I don't object to changes to these comments but we could easily wind up wasting a lot of time with comment editing if we try to get strict about requiring specific rules.

I just did a quick search & replace, I don't think we should enforce it as a rule.

msft-jlange commented on kernel/src/cpu/apic.rs:
```diff
@@ -249,6 +249,8 @@ impl LocalApic {
         Some(calling_area)
     }
 
+    /// Attempts to deliver the specified IRQ into the specified guest CPU
```
We should distinguish this from queueing an interrupt.  The key here is that this function attempts to deliver the specified interrupt so it will be observed immediately upon the next guest entry, and will fail if the guest state is such that interrupt delivery would need to be deferred (e.g. due to disabled interrupts or interrupt priority).  I'm not sure how to state this succinctly in a comment, but it would be helpful.

00xc replied:
Updated the doc text to include this.

msft-jlange commented:
@00xc, this has been pending update for a while.  Would you like me to take over the PR and complete it, or did you want to continue to update it yourself?

00xc committed:
cpu/apic: multiple miscellaneous cleanups

Add documentation for functions that simply return a boolean value to
improve clarity. While we are at it, remove the `pub` qualifier from
functions that are not used outside the APIC module. Clean up as well
the formatting in some comments.

Signed-off-by: Carlos López <carlos.lopezr4096@gmail.com>

00xc committed:
cpu/apic: remove two unnecessarily mutable variables

Use constructor methods and make two mutable variables immutable.

Signed-off-by: Carlos López <carlos.lopezr4096@gmail.com>

00xc commented:
> @00xc, this has been pending update for a while. Would you like me to take over the PR and complete it, or did you want to continue to update it yourself?

Should be good to go now

msft-jlange commented:
I still don't like all of the double- to single-space changes because it introduces a lot of noise into the diff (it makes it harder to see the substantive changes because they're mixed in with a lot of whitespace changes) but the rest of the change looks good.
